client/src/elements/schema-editor/schema-editor-array.js

```diff
@@ -34,6 +34,16 @@ export class SchemaEditorArray {
   }
 
   dataChanged() {
+    // create a new dataItemsSchema array
+    // this array will hold the schema per data array item
+    // and will mirror any changes to the array
+    // this way, we do not need to calculate the schemas for the data
+    // on all changes. For addElement we already know the schema beforehand e.g.
+    this.dataItemsSchemas = this.data
+      .map(item => {
+        return this.getSchemaForData(item);
```

Comment:
just a question for understanding: does map filter null values? or do they get handled anywhere?

Reply:
`map` just executes the function for every array entry and returnes a new array containing the return values of the function. So no, it doesn't filter.
But there should be no `null` values in this array (except if someone messes with the data in the database directly) as they should not get saved if they do not match the schema for the entry. If they match the schema, then we do not have a problem here, as `this.getSchemaForData(item)` would return a schema in this case.

Reply:
I thought so, that practically it should not happen that we have a null value. I'm just a bit precautious when null values are returned :)